### DISCUSSION TITLE
resolv hostnames from IP-Adresse

### DISCUSSION THREAD

**Discussion by Dex321x** [AdguardTeam/AdGuardHome#5951] [Q&A]
Hello,

Recently, Adguard Home can't show my the device names. It only shows the IP addresses. In the past weeks it was working from time to time. But now it won't work anymore. Tried restarting routers etc. but didn't help.

All devices are plugged into an Asus RT-AC86U router with IP 192.168.1.1
He also manages the DHCP.
Adguard Home is running on an PI with the address 192.168.1.22

Adguard Home DNS Settings:
![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/60154621/84af9fe8-d51e-4608-8827-c63a753e4077)

Asus Router DHCP Settings:
![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/60154621/24006e75-6b0a-457c-9435-e09a5b812de4)

I tried adding this to upstream dns server (found in the [wiki](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-rdns)), but didn't do anything. It seems to be outdated too in the latest version (if i understood it correct)
```
[/in-addr.arpa/]192.168.1.1
[/ip6.arpa/]192.168.1.1
```

I'm out of ideas what the issue is. Any ideas what I should try?

Thanks, Dex

**Comment by ainar-g** [comment]
Your configuration should work.  The one with a single private reverse DNS server.  Which command are you using to test it?

**Comment by Dex321x** [comment]
Thanks for your reply. 
I didn't use any command. I just looked at the Adguard statistic charts and there it just showed the IP address instead of the name. 

Your answer made me come back to check about that (I gave up already) and now in the Adguard statisics nearly every device is displayed with the correct name. It's strange, but now it seems to work again.

**Reply by ainar-g** [reply]
Perhaps there was a bug we've since fixed.  If you are able to reproduce the issue reliably, please file a bug report in Issues.

**Comment by omalez21** [comment]
I have the exact same issue on my end as well, currently on version v0.107.35 , my config matches the screenshot OP has for AdGuard Home. I have AdGuard home setup on my raspberry pi and tried to install the dnsutils to run a "dig -x" on local ip address and can see it's able to resolve the hostname. 

initially, I had noticed under the clients' section in AdGuard home and would notice under the "source" column it was showing "rDNS" and the hostname was present, but now its reverted to the "source" of ARP and can't seem to figure out why.

**Reply by omalez21** [reply]
looking further, the only thing I can see is under the doc for configuration:
[AdGuardHome Configuration Docs](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file)

there is an option called `resolve_clients` that looks to be similar if not the same as the one on the AdminGUI `enable reverse resolving of clients' IP addresses`, however when I try to stop the AdGuardHome service and then use sudo nano to edit the config file to add this option because it's missing in mine it seems to just remove itself when I start AdGuardHome back up again.

could that be related?

**Comment by Just2Confuse** [comment]
Bug since the last version.
https://github.com/AdguardTeam/AdGuardHome/issues/6046

**Comment by TNOTA** [comment]
Yep, updated to latest version yesterday and now Adguard will not resolve hostnames.

**Comment by TNOTA** [comment]
Rolled back to v0.107.34 and everything works again.

**Reply by omalez21** [reply]
Are you able to check the configuration file and see if `resolve_clients` exists? And what the value is?

If your on raspberry like myself the config file is /opt/AdGuardHome/AdGuardHome.yaml

**Reply by TNOTA** [reply]
I am running it on a Mac. I don't see that in either the yaml from the latest version or the prior version.

**Reply by omalez21** [reply]
Interesting, thank you for checking.

**Comment by Coffee0297** [comment]
still an issue in latest
